### PR TITLE
Port latest Normative changes from tc39/proposal-temporal.

### PR DESCRIPTION
**Pull request by 12wrigja**
This also bumps up the Test262 test cases to commit  3dbf940c9bff08b05a87e34e98864887550bda8d to match the current state of the tc39/proposal-temporal repo.

### CODE REVIEW

**Comment by 12wrigja** [comment]
Fine by me - thanks for the heads up.

**Comment by justingrant** [comment]
> Do you mean [tc39/proposal-temporal#1878](https://github.com/tc39/proposal-temporal/issues/1878) ?

That's the root cause spec bug, but #1861 has links to Test262 changes that need to be reverted.

**Comment by 12wrigja** [comment]
I'm re-using this PR to port over the various changes that have landed recently to the non-prod polyfill.

File: lib/zoneddatetime.ts [hunk 1]
```diff
@@ -212,9 +203,29 @@ export class ZonedDateTime implements Temporal.ZonedDateTime {
...
+      ['millisecond', 0],
+      ['minute', 0],
+      ['month', undefined],
+      ['monthCode', undefined],
+      ['nanosecond', 0],
+      ['second', 0],
+      ['year', undefined],
+      ['offset'],
+      ['timeZone']
+    ];
```
Comment (justingrant) [inline lib/zoneddatetime.ts:219]: In other types, the `entries` variable is typed (via a type annotation) as something like this:

```js
const entries: [keyof Temporal.ZonedDateTimeLike, 0 | undefined][]
```

This removes the need for the `as any` below.

usually these `entries` values are typed `as const` or with a type annotation on

**Review by justingrant — COMMENTED** [review]
Looks good so far. One minor type annotation suggestion. 

Probably should change the PR title to match the wider scope. 

Not sure we want to delete PlainMonthDay demitasse tests even if they exist in Test262. Let's discuss pros/cons of this.

**Comment by ptomato** [comment]
Agreed, I would not remove the PlainMonthDay tests.

**Comment by 12wrigja** [comment]
Done - I've brought back and fixed up the single failing testcase.

**Review by ptomato — APPROVED** [review]

**Pull request merged**